### PR TITLE
🎱 859 Create UI and instrument it

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds the UI portion for the pokeshop project. It has been added as a mono repo the same way we handle the trace test one.
It also adds instrumentation to the front end by adding the open telemetry tools.

## Changes

- Front-end implementation
- Instrumentation
- Backend propagation tooling.

## Fixes

- https://github.com/kubeshop/tracetest/issues/859

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

https://www.loom.com/share/53477126e7044f57af8009e2d51d7efb

### CODE REVIEW

File: api/src/api.ts [hunk 1]
```diff
@@ -36,7 +41,16 @@ async function startApp() {
     routeSetup(router);
   }
 
-  app.use(instrumentRoute()).use(bodyParse()).use(KoaLogger()).use(router.routes()).use(router.allowedMethods());
+  app
+    .use(cors())
```
Comment (xoscar) [inline api/src/api.ts:45]: Adding cors, static ui and moar

File: api/src/telemetry/tracing.ts [hunk 2]
```diff
@@ -9,10 +9,13 @@ import { IORedisInstrumentation } from '@opentelemetry/instrumentation-ioredis';
 import { PgInstrumentation } from '@opentelemetry/instrumentation-pg';
 import { AmqplibInstrumentation } from '@opentelemetry/instrumentation-amqplib';
 import { SpanStatusCode } from '@opentelemetry/api';
+import { B3Propagator } from '@opentelemetry/propagator-b3';
 
 // Make sure all env variables are available in process.env
 dotenv.config();
 
+api.propagation.setGlobalPropagator(new B3Propagator());
```
Comment (xoscar) [inline api/src/telemetry/tracing.ts:17]: OTEL propagator

File: web/src/hooks/usePokemonCrud.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { useQuery, useMutation, useQueryClient } from 'react-query';
+import PokemonGateway from '../gateways/pokemon';
+
+const CACHE_KEY = 'pokemonList';
+const FEATURED_CACHE_KEY = `${CACHE_KEY}/featured`;
+
+const usePokemonCrud = () => {
```
Comment (xoscar) [inline web/src/hooks/usePokemonCrud.ts:7]: Using react query

File: web/src/types/generated.ts [hunk 4]
```diff
@@ -0,0 +1,285 @@
+/**
+ * This file was auto-generated by openapi-typescript.
+ * Do not make direct changes to the file.
+ */
+
+export interface paths {
```
Comment (xoscar) [inline web/src/types/generated.ts:6]: Generated types form the open api specs

File: web/src/utils/request.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+const request = async <T>({
+  url = '',
+  method = 'GET',
+  body,
+  queryParams = {},
+  headers = {
+    'content-type': 'application/json',
+  },
+}: IRequestParams): Promise<T> => {
```
Comment (xoscar) [inline web/src/utils/request.ts:17]: Request util

**Comment by mathnogueira** [comment]
It would be good to have feedback from someone from the UI team. @cescoferraro @jorgeepc

File: web/package.json [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    "@opentelemetry/propagator-b3": "^1.4.0",
+    "@opentelemetry/sdk-trace-base": "^1.4.0",
+    "@opentelemetry/sdk-trace-web": "^1.4.0",
+    "@testing-library/jest-dom": "^5.16.4",
+    "@testing-library/react": "^13.3.0",
+    "@testing-library/user-event": "^13.5.0",
+    "@types/jest": "^27.5.2",
+    "@types/node": "^16.11.43",
+    "@types/react": "^18.0.15",
+    "@types/react-dom": "^18.0.6",
```
Comment (jorgeepc) [inline web/package.json:20]: dev dependencies?

**Review by jorgeepc — APPROVED** [review]
🔥 Amazing work @xoscar

File: web/src/utils/tracer.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+});
+
+provider.addSpanProcessor(
+  new BatchSpanProcessor(
+    new ZipkinExporter({
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      serviceName: 'pokeshop-ui',
+      url: `${process.env.REACT_APP_ZIPKIN_URL}/api/v2/spans`,
```
Comment (jorgeepc) [inline web/src/utils/tracer.ts:24]: Curious, where are we setting this env `REACT_APP_ZIPKIN_URL`? And what is this endpoint?

**Pull request merged**